integration/startup_test.go

```diff
@@ -137,3 +142,101 @@ func TestSnapshotterSystemdStartup(t *testing.T) {
 		})
 	}
 }
+
+func TestStartWithDefaultConfig(t *testing.T) {
+	defaultConfigToml, err := os.ReadFile(defaultConfigFileLocation)
+	if err != nil {
+		t.Fatalf("error fetching example toml")
+	}
+
+	sh, c := newSnapshotterBaseShell(t)
+	defer c()
+
+	rebootContainerd(t, sh, getContainerdConfigToml(t, false), string(defaultConfigToml))
+	// This will error internally if it fails to boot. If it boots successfully,
+	// the config was successfully parsed and snapshotter is running
+}
+
+// TestStartWithoutConfig checks that SOCI can start with specified config paths
+func TestStartWithConfigPaths(t *testing.T) {
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+
+	snapshotterSocket := "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+
+	tests := []struct {
+		name        string
+		defaultPath bool
+		fileExists  bool
+		expectErr   bool
+	}{
+		{
+			name:        "should start if config file is present in default location",
+			defaultPath: true,
+			fileExists:  true,
+		},
+		{
+			name:        "should start if config file is not present in default location",
+			defaultPath: true,
+			fileExists:  false,
+		},
+		{
+			name:        "should start if config file is present in non-default specified location",
+			defaultPath: false,
+			fileExists:  true,
+		},
+		{
+			name:        "should not start if config file is not present in non-default specified location",
+			defaultPath: false,
+			fileExists:  false,
+			expectErr:   true,
```

Comment:
I think we should try to use error matchers (or strings that should appear in the error message) rather than bools. It's easy for something completely unrelated to fail and not notice because we look for any error.

Reply:
Yeah good point. Let me address this.